### PR TITLE
topoplgy: add tgl rt715-rt711-rt1308 support

### PR DESCRIPTION
**Pull request by bardliao**
Add tgl rt715-rt711-rt1308 support. But keep the PCM order as
Headphone, Headset mic, Speakers, Microphones.

Signed-off-by: Bard Liao <yung-chuan.liao@linux.intel.com>

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
CI failure on ICL is unrelated.

File: tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4 [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+#
+# DAIs configuration
+#
+
+dnl DAI_ADD(pipeline,
+dnl     pipe id, dai type, dai_index, dai_be,
+dnl     buffer, periods, format,
+dnl     deadline, priority, core, time_domain)
+
+# playback DAI is ALH(SDW0 PIN2) using 2 periods
```
Comment (plbossart) [inline tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4:99]: SDW1 Pin2

File: tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4 [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+# Buffers use s24le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	1, ALH, 0x102, SDW1-Playback,
+	PIPELINE_SOURCE_1, 2, s24le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# capture DAI is ALH(SDW0 PIN2) using 2 periods
```
Comment (plbossart) [inline tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4:106]: SDW1 Pin3

File: tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4 [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+# Buffers use s24le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	2, ALH, 0x103, SDW1-Capture,
+	PIPELINE_SINK_2, 2, s24le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is ALH(SDW1 PIN2) using 2 periods
```
Comment (plbossart) [inline tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4:113]: SDW2 Pin2

File: tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4 [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+# Buffers use s24le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	3, ALH, 0x202, SDW2-Playback,
+	PIPELINE_SOURCE_3, 2, s24le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# capture DAI is ALH(SDW3 PIN2) using 2 periods
```
Comment (plbossart) [inline tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4:120]: SDW0 Pin2

File: tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4 [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+PCM_PLAYBACK_ADD(HDMI2, 6, PIPELINE_PCM_7)
+PCM_PLAYBACK_ADD(HDMI3, 7, PIPELINE_PCM_8)
+
+#
+# BE configurations - overrides config in ACPI if present
+#
+
+#ALH dai index = ((link_id << 8) | PDI id)
+#ALH SDW0 Pin2 (ID: 0)
+DAI_CONFIG(ALH, 2, 0, SDW1-Playback,
```
Comment (plbossart) [inline tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4:164]: this doesn't seem right? this should be capture.

File: tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4 [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+	ALH_CONFIG(ALH_CONFIG_DATA(ALH, 2, 48000, 2)))
+
+#ALH SDW0 Pin3 (ID: 1)
+DAI_CONFIG(ALH, 3, 1, SDW1-Capture,
```
Comment (plbossart) [inline tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4:168]: could it be 0x103?

**Review by plbossart — CHANGES_REQUESTED** [review]
need to double-check the ALH pin assignments? The last part doesn't seem right to me.

File: tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4 (hunk 6 above)
Comment (bardliao) [reply tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4]: Yes, it should be 0x103

File: tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4 (hunk 5 above)
Comment (bardliao) [reply tools/topology/sof-tgl-rt715-rt711-rt1308-hdmi.m4]: It should be 0x102 instead

**Comment by bardliao** [comment]
Thanks @plbossart All comments are addressed.

**Comment by lgirdwood** [comment]
@mengdonglin @bardliao I've marked for v1.6 so we can pull it int v1.6 final. I will merge today and cherry-pick to v1.6 branch once @plbossart approves.

**Review by plbossart — APPROVED** [review]
Looks good now, thanks @bardliao

**Pull request merged**